### PR TITLE
Pstats merge #35

### LINKED ISSUE #35: merge-pstats assumes stats are always overlapping in time
Hi, and thanks once again for a great library.

It seems that `merge-pstats` assumes that its inputs are always overlapping in time.
For example:

```clojure
(test/deftest merge-discrete-events
  (test/testing "Merge discrete events"
    (let [[_ ps0] (profiled {} (p :foo "foo"))
          _ (Thread/sleep 100) ; Majority of time spent outside of profiling
          [_ ps1] (profiled {} (p :foo "foo"))
          merged @(tufte/merge-pstats ps0 ps1)]
      (pprint/pprint merged)
      (is (< (get-in merged [:clock :total]) 1e8))))) ; Fails
```

This test will fail and print:
```clojure
{:clock {:t0 2109412625335, :t1 2109513733727, :total 101108392},
;                                                     ^^^ Incorrect total, includes (Thread/sleep 100)
 :stats
 {:foo
  {:min 181,
   :mean 501.0,
   :mad-sum 640.0,
   :p99 821,
   :n 2,
   :p90 821,
   :max 821,
   :mad 320.0,
   :p50 821,
   :sum 1002,
   :p95 821}}}
```

I hit this error while trying to write a simple aggregate handler. I suppose that in nested cases of `profiled` it makes sense to assume overlapping, but not when you have multiple discrete events. I'm not quite sure how to solve this correctly. Or am I doing something wrong?

Regards.

### PR DESCRIPTION
**Pull request by ivarref**
Fixes #35

### CODE REVIEW

**Comment by ptaoussanis** [comment]
Hi Ivar, thanks for this.

Please note that I haven't had an opportunity to look at this yet, and am expecting I probably won't have the opportunity for the next 2-3 weeks at least.

In the meantime, could you perhaps provide a motivating problem for #35 - just to make sure we're on the same page? Under what circumstances are you expecting to want to merge stats that didn't occur during the same time window?

**Comment by ptaoussanis** [comment]
Thanks for the extra info.

Use case makes complete sense. What I'd normally suggest in this case would be something like the following:

You keep a buffer of stats for a configurable window period of length n. (Often 5-15 minutes, etc.). During that time, all profiling calls (from all endpoints) will accumulate their stats into the shared buffer.

Then at the end of the window period- you drain the buffer, merge the stats, and push to your db/dashboard/etc.

Basically you'll get a high-level aggregated view of your whole system's performance over the last periods - and all stats exist within a known window t1-t0.

Does that make sense?
